geode-core/src/main/java/org/apache/geode/internal/net/NioFilter.java

```diff
@@ -35,7 +35,7 @@
    * Be sure to call close() on the returned {@link ByteBufferSharing}. The best way to do that is
    * to call this method in a try-with-resources statement.
    */
-  ByteBufferSharing wrap(ByteBuffer buffer) throws IOException;
+  ByteBufferSharing wrap(ByteBuffer buffer, final SocketChannel channel) throws IOException;
```

Comment:
`wrap()` now must take a `SocketChannel` parameter because `wrap()` may have to write to the channel before returning to the caller.

Reply:
Isn't the `NioFilter` unique to each `Socket`/`SocketChannel`, why isn't this a member?

Reply:
reverted this change